### PR TITLE
packio: add 2.1.0

### PR DESCRIPTION
**Pull request by qchateau**
Specify library name and version:  **packio/2.1.0**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by uilianries — APPROVED** [review]

File: recipes/packio/all/test_package/CMakeLists.txt [hunk 1]
```diff
@@ -9,7 +9,7 @@ if (PACKIO_VERSION VERSION_LESS "1.2.0")
 elseif (PACKIO_VERSION VERSION_LESS "2.0.0")
     add_executable(main 1.2.x.cpp)
 else ()
-    add_executable(main 2.0.x.cpp)
+    add_executable(main 2.0.x-2.1.x.cpp)
```
Comment (prince-chrismc) [inline recipes/packio/all/test_package/CMakeLists.txt:12]: Can you please rename to test_package.cpp or example.cpp ? This way it does not need to change in the near future.

Comment (qchateau) [reply recipes/packio/all/test_package/CMakeLists.txt:12]: Maybe "latest.cpp" ?

Comment (prince-chrismc) [reply recipes/packio/all/test_package/CMakeLists.txt:12]: My suggestion is based on the existing ones. 

It's not a blocker but it's odd to change this for a bump

**Review by SSE4 — DISMISSED** [review]

File: recipes/packio/all/test_package/CMakeLists.txt (hunk 1 above)
Comment (qchateau) [reply recipes/packio/all/test_package/CMakeLists.txt]: I agree it should not change in the near future.
I went for "latest.cpp", it makes more sense to me considering there are 1.2.x.cpp and 1.0.x-1.1.x.cpp in the same directory.

**Review by prince-chrismc — APPROVED** [review]

**Review by intelligide — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**